superset/security/manager.py

```diff
@@ -1097,6 +1105,12 @@ def get_user_by_username(
     def get_anonymous_user(self) -> User:  # pylint: disable=no-self-use
         return AnonymousUserMixin()
 
+    def get_user_roles(self) -> List[Role]:
+        if g.user.is_anonymous:
+            public_role = current_app.config.get("AUTH_ROLE_PUBLIC")
+            return [self.get_public_role()] if public_role else []
+        return g.user.roles
```

Comment:
should add a test for both anonymous and standard users (or maybe move a test that exists already in another file?)